joomla/joomla-cms
# [4.0] Namespace com contact

Pull Request for Issue # .

### Summary of Changes
As title says, this PR convert com_contact component to namespace MVC for Joomla 4.


### Testing Instructions
Overall, test and make sure it works in the none namespace version of the extension

1. Install a copy of my branch https://github.com/joomdonation/joomla-cms/archive/namespace_com_contact.zip (to avoid problem with patchtesters component not apply patch properly)

2. Try to add/edit/publish/unpublish/archive contact from administrator area of the site. Make sure it is working as expected

3. Access to frontend of the site, look at **All Front End Views** module, access to all the menu items 
- Contact Categories
- Contact Single Category
- Single Contact
- Featured Contacts
- Try to send email from contact form

Make sure it is all working as expected. Note on local computer, you might get the error **500 Could not instantiate mail function.**, it is unrelated to this PR, so ignore it (you would get the same error on none namespace version if you want to confirm)

### Additional comments
In almost every core extensions, there are helper classes like Category Helper, Association Helper, Router Helper... Namespace these classes are easy, but we need to find a way to call the namespace classes directly in the places which it is used instead of create a none namespace class to extends that namespace class (or registering aliases). For now, I leave these classes as none namespace until we agree on a solution
